src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Mapping/RazorMapToDocumentRangesEndpoint.cs

```diff
@@ -45,35 +45,42 @@ public Uri GetTextDocumentIdentifier(RazorMapToDocumentRangesParams request)
             return null;
         }
 
-        if (request.Kind != RazorLanguageKind.CSharp)
+        var codeDocument = await documentContext.GetCodeDocumentAsync(cancellationToken).ConfigureAwait(false);
+        IRazorGeneratedDocument? generatedDocument = request.Kind switch
         {
-            // All other non-C# requests map directly to where they are in the document.
-            return new RazorMapToDocumentRangesResponse()
-            {
-                Ranges = request.ProjectedRanges,
-                HostDocumentVersion = documentContext.Snapshot.Version,
-            };
+            RazorLanguageKind.CSharp => codeDocument.GetCSharpDocument(),
+            RazorLanguageKind.Html => codeDocument.GetHtmlDocument(),
```

Comment:
I was on the fence here because if we want to be "correct" we need to populate spans and ranges. However in practice this information isn't used in HTML

Reply:
We should probably just decide what we're doing about this at some point, and either remove html mappings, or use them.

Reply:
FWIW, it's wasteful to call `GetCodeDocumetnAsync(...)` in the cases where we'd just throw it away. IMO, if `request.Kind == RazorLanguageKind.Razor`, we should exit early rather than forcing the Razor file to be compiled.

Reply:
I'll add a check earlier for `Razor`